lib/Sema/TypeCheckType.cpp

```diff
@@ -1654,13 +1710,14 @@ bool TypeChecker::validateType(TypeLoc &Loc, TypeResolution resolution,
     type = resolution.resolveType(Loc.getTypeRepr(), options);
     if (!type) {
       type = ErrorType::get(Context);
-
       // Diagnose types that are illegal in SIL.
     } else if (options.contains(TypeResolutionFlags::SILType)
                && !type->isLegalSILType()) {
       diagnose(Loc.getLoc(), diag::illegal_sil_type, type);
       Loc.setInvalidType(Context);
       return true;
+    } else if (validateAutoClosureAttributeUse(*this, Loc, type, options)) {
+      type = ErrorType::get(Context);
```

Comment:
Can we drop the `@autoclosure` rather than going all the way to `ErrorType`?

Reply:
I'd like for parameter declaration to be marked invalid which would reset its type to ErrorType anyway...

Reply:
Okay!